src/crimson/common/tri_mutex.cc

```diff
@@ -181,12 +183,11 @@ bool tri_mutex::is_acquired() const
   }
 }
 
-void tri_mutex::wake()
+void tri_mutex::wake(type_t type)
 {
   LOG_PREFIX(tri_mutex::wake());
   DEBUGDPP("", *this);
-  assert(!readers && !writers && !exclusively_used);
-  type_t type = type_t::none;
+  assert(!exclusively_used);
```

Comment:
I'm not sure we want to remove the original assertion. This is the only way which we can verify that when allowing write, no other reads are in progress (or vice versa)

Reply:
Just re-added assertions to do the same validation.